app/helpers/application_helper.rb

```diff
@@ -1,7 +1,7 @@
 module ApplicationHelper
 
   def admin?
-    !current_user.nil? and current_user.admin?
+    user_signed_in? and current_user.admin?
```

Comment:
Where'd this user_signed_in come from? Can this be used in other files rather than the current_user.nil?

Reply:
this is coming from a helper provided by Devise (the authen gem being used), As such, it is not available to the controllers themselves, but is a view helper. As far as I can tell.

Reply:
Nope. I was wrong. It is accessible in controllers. And used now.